apps/admin/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+
+import {
+  flexRender,
+  getCoreRowModel,
+  getPaginationRowModel,
+  useReactTable,
+  getSortedRowModel,
+  getFilteredRowModel,
+} from '@tanstack/react-table';
+
+import { rankItem } from '@tanstack/match-sorter-utils';
+
+import DebouncedInput from './DebouncedInput';
+
+function Table({ tableData, columns }) {
+  const [data, setData] = useState(() => [...tableData]);
+  const [sorting, setSorting] = useState([]);
+  const [globalFilter, setGlobalFilter] = useState('');
+
+  const fuzzyFilter = (row, columnId, value, addMeta) => {
+    // Rank the item
+    const itemRank = rankItem(row.getValue(columnId), value);
+
+    // Store the itemRank info
+    addMeta({
+      itemRank,
+    });
+
+    // Return if the item should be filtered in/out
+    return itemRank.passed;
+  };
+
```

Comment:
what is this doing?

Reply:
I dont know

![Alt Text](https://media.giphy.com/media/vFKqnCdLPNOKc/giphy.gif)